flang/include/flang/Optimizer/CodeGen/CGOps.td

```diff
@@ -82,6 +82,51 @@ def fircg_XEmboxOp : fircg_Op<"ext_embox", [AttrSizedOperandSegments]> {
   }];
 }
 
+// Extended rebox operation.
+def fircg_XReboxOp : fircg_Op<"ext_rebox", [AttrSizedOperandSegments]> {
+  let summary = "for internal conversion only";
+
+  let description = [{
+    Prior to lowering to LLVM IR dialect, a non-scalar non-trivial rebox op will
+    be converted to an extended rebox. This op will have the following sets of
+    arguments.
+
+       - box: The box being reboxed.
+       - newShape: A vector that is the new runtime shape for the array
+       - newShift: A vector that is the new runtime origin of the first element.
+         The default is a vector of the value 1.
+       - slice: A vector of triples that describe an array slice.
+       - subcomponent: A vector of indices for subobject slicing.
+       - sliceOrigins: A vector that is the new runtime origin of the slice operand
+
+    The box argument is mandatory, the other arguments are optional.
+  }];
+
+  let arguments = (ins
+    fir_BoxType:$box,
+    Variadic<AnyIntegerType>:$newShape,
+    Variadic<AnyIntegerType>:$newShift,
+    Variadic<AnyIntegerType>:$slice,
+    Variadic<AnyCoordinateType>:$subcomponent,
+    Variadic<AnyIntegerType>:$sliceOrigins
+  );
+  let results = (outs fir_BoxType);
+
+  let assemblyFormat = [{
+    $box (`(`$newShape^`)`)? (`origin` $newShift^)? (`[`$slice^`]`)?
+      (`path` $subcomponent^)? (`sliceorigins` $sliceOrigins^)? attr-dict
+      `:` functional-type(operands, results)
+  }];
```

Comment:
I intentionally don't give the codegen ops a syntax because I didn't want anyone using them directly. (Not having a nice syntax makes it harder for someone to "try them out".)

Reply:
I am not sure I get the comment. Do you mean there should not be an `assemblyFormat` definition ?
The other codegen ops also have `assemblyFormat`:
https://github.com/flang-compiler/f18-llvm-project/blob/bb5efb8ac6165f38ec622563353ee1b800108ec7/flang/include/flang/Optimizer/CodeGen/CGOps.td#L61-L65
https://github.com/flang-compiler/f18-llvm-project/blob/bb5efb8ac6165f38ec622563353ee1b800108ec7/flang/include/flang/Optimizer/CodeGen/CGOps.td#L122-L126

Reply:
Argh. I forgot I had added them. My bad.